addons/web/static/src/core/tags_list/badge_tag.xml

```diff
@@ -7,8 +7,8 @@
                 <div class="o_tag_badge_text o_badge_text" t-out="this.props.text"/>
             </t>
             <t t-if="this.props.onDelete">
-                <a class="o_delete d-flex align-items-center opacity-100-hover ps-1 opacity-75" href="#" tabindex="-1" aria-label="Delete" data-tooltip="Delete" t-on-click.stop.prevent="() => this.props.onDelete()">
-                    <i class="oi oi-close align-text-center"/>
+                <a class="o_delete d-block ps-1 opacity-75 opacity-100-hover" href="#" tabindex="-1" aria-label="Delete" data-tooltip="Delete" t-on-click.stop.prevent="() => this.props.onDelete()">
```

Comment:
`.badge` inside `.o_field_many2many_tags` are flex by default so `.o_delete` will be automatically `display: block;`. Therefore I think we can remove the display utility class here.